### ISSUE TITLE
bug: The order of the alert buttons on iOS is reversed

### ISSUE THREAD

**Issue by yudateNoriyuki** [ionic-team/ionic-framework#26367]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [X] v4.x
- [X] v5.x
- [X] v6.x
- [ ] Nightly

### Current Behavior

When button names are long, the order of alert buttons in iOS is the reverse of that in Android.

### Expected Behavior

The order of iOS should be the same as Android.

### Steps to Reproduce

1. I see an alert for https://ionicframework.com/docs/ja/api/alert#buttons
2. Edit the html and name the 'Cancel' button 'CancelCancel (Buttons are aligned vertically)
3. Compare iOS and MD.

## iOS
![image](https://user-images.githubusercontent.com/50348650/204438354-b71ac9ac-ad6c-48d7-9d98-3eb665363127.png)

## MD
![image](https://user-images.githubusercontent.com/50348650/204438433-99bc29cc-a419-4458-9296-79ce0413b303.png)

### Ionic Info

-

### Additional Information

plz look at 

### MD
https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/alert/alert.md.vars.scss#L122
### iOS
https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/alert/alert.ios.vars.scss#L129

The differences in properties affect this sort order.

**Comment by liamdebeasi** [comment]
Thanks for the report. I can reproduce this behavior.

For stacked buttons in Material Design alerts, the confirming action should appear above the cancelling action: https://www.material.io/components/dialogs#anatomy (See "Stacked full-width buttons" at the bottom). This is what is happening right now in Ionic: https://stackblitz.com/edit/angular-h8ajiu?file=src%2Fapp%2Fexample.component.ts,src%2Fapp%2Fexample.component.html

For stacked buttons in iOS alerts, the cancelling action should always appear last. The non-cancelling actions appear in the order they were passed. Here is an example of native iOS:

![image](https://user-images.githubusercontent.com/2721089/204581388-0e212a53-4e02-4d1d-9709-11936ca46563.png)

Related Swift UI code:


```swift
.alert("Important message", isPresented: $showingAlert) {
  Button("Cancel Cancel", role: .cancel) { }
  Button("Regular Button") { }
  Button("Delete Delete", role: .destructive) { }
}
```

This is different from inline buttons where the cancelling action should always appear first. 

------

To summarize, we need to make the following change:

1. Stacked buttons on iOS should always render the "cancel" button last. All other buttons should render the order in which they were presented.

**Comment by yudateNoriyuki** [comment]
@liamdebeasi 
Thanks for the reply.
If I keep waiting, will the iOS stack button be fixed?

**Comment by liamdebeasi** [comment]
We are interested in fixing this issue. However, I do not have a timeline of when it will be fixed.
